### PR TITLE
Add Reth to execution clients suggested for installation

### LINKED ISSUE #6171: Add Reth to execution clients suggested for installation
## Description

The [installation](https://lighthouse-book.sigmaprime.io/installation.html) mentions a few execution clients.
Reth is in production and it would be nice to include it as a suggestion for use for a full Rust node.

Edit: there are a few other places that mention the EL so Reth may be added there as well.

## Present Behaviour

Reth is not mentioned in the list of clients in the installation page.

## Expected Behaviour

Hyperlink the GitHub repo to Reth in the installation page among the other execution clients.

### PR DESCRIPTION
**Pull request by Braqzen**
## Issue Addressed

Closes #6171 

## Proposed Changes

The following pages now include Reth among other execution clients

- installation.md
- merge-migration.md
- run_a_node.md

### CODE REVIEW

File: book/src/run_a_node.md [hunk 1]
```diff
@@ -26,6 +26,7 @@ per beacon node. The reason for this is that the beacon node *controls* the exec
 - [Besu](https://besu.hyperledger.org/en/stable/public-networks/get-started/connect/mainnet/)
 - [Erigon](https://github.com/ledgerwatch/erigon#beacon-chain-consensus-layer)
 - [Geth](https://geth.ethereum.org/docs/getting-started/consensus-clients)
+- [Reth](https://reth.rs/run/run-a-node.html)
```
Comment (chong-he) [inline book/src/run_a_node.md]: I think for the general users looking to setup a node, we can link directly to the page of running a mainnet/testnet Reth node here.

```suggestion
- [Reth](https://reth.rs/run/mainnet.html)
```

**Review by chong-he — COMMENTED** [review]
Looks great, just a small suggestion here

**Review by michaelsproul — APPROVED** [review]
LGTM! Thanks!

**Comment by michaelsproul** [comment]
@mergify queue

**Pull request merged**